### PR TITLE
Add code owners for IDE-related directories

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Review by bnbarham — APPROVED** [review]
Just include me in all please 🙏

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -61,6 +61,8 @@
 /include/swift/DependencyScan                      @artemcm
 /include/swift/Driver                              @artemcm
 # TODO: /include/swift/IRGen/
+/include/swift/IDE/                                @ahoppen @rintaro
+/include/swift/Refactoring                         @ahoppen @bnbarham
```
Comment (AnthonyLatsis) [inline .github/CODEOWNERS:65]: I see you’ve added `test/Index`. Were the indexing sources purposely omitted?

**Review by AnthonyLatsis — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (ahoppen) [reply .github/CODEOWNERS:66]: I just missed them. Thanks for noticing

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Pull request merged**